### PR TITLE
fix(shared,nextjs): Tests for `isHttpOrHttps` utility

### PR DESCRIPTION
**Pull request by panteliselef**
## Type of change

- [] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [x] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [x] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

This PR adds tests for `isHttpOrHttps` utility
and  isHttpOrHttps now uses regex instead of .startsWith()

### CODE REVIEW

**Review by SokratisVidros — APPROVED** [review]

File: packages/shared/src/utils/proxy.test.ts [hunk 1]
```diff
@@ -24,6 +24,24 @@ describe('isProxyUrlRelative(key)', () => {
   });
 });
 
+describe('isHttpOrHttps(key)', () => {
+  it('returns true if the proxyUrl starts with `http://`', () => {
```
Comment (SokratisVidros) [inline packages/shared/src/utils/proxy.test.ts:28]: 🙃 Check also it.each. 

Please add a test case for empty string.

Comment (panteliselef) [reply packages/shared/src/utils/proxy.test.ts]: thank you for the suggestions. I added them
